reana_cluster/backends/kubernetes/k8s.py

```diff
@@ -0,0 +1,510 @@
+    def cluster_type(self):
+        return self.__cluster_type
+
+    @property
+    def cluster_type(self):
+        return self.cluster_conf
+
+    @property
+    def current_config(self):
+        # TODO: Check what conf holds when `load_kube_config()` hasn't been run
+        if configuration is None:
+            # TODO: Raise somekind of "NotInitialized" error is
+            # load_kube_config() has not been run.
+            pass
+        # TODO: Probably better to return only relevant data not the whole conf
+        return configuration
+
+    @classmethod
+    def generate_configuration(cls, cluster_spec):
+        """Generate Kubernetes backend configuration files to init REANA cluster.
+
+        :param cluster_spec: Dictionary representing complete REANA
+            cluster spec file.
+        """
+
+        # Setup an Environment for Jinja
+        env = Environment(
+            loader=FileSystemLoader(
+                cls._conf['templates_folder']),
+            keep_trailing_newline=False
+        )
+
+        # Define where are backend conf params needed when rendering templates.
+        be_conf_params_fp = cls._conf['templates_folder'] + '/config.yaml'
+
+        try:
+            with open(be_conf_params_fp) as f:
+
+                # Load backend conf params
+                backend_conf_parameters = yaml.load(f.read())
+
+                # Load template combining all other templates from
+                # templates folder
+                # TODO: Better to combine templates or populated templates in
+                # Python code for improved extensibility?
+                # Just drop a .yaml template and add necessary to config.yaml
+                # without changing anything?
+                # template = env.get_template('deployments/complete.yaml')
+                template = env.get_template('complete.yaml')
+
+                components = cluster_spec['components']
+                server_img = components['reana-server']['image']
+                jc_img = components['reana-job-controller']['image']
+                wfc_img = components['reana-workflow-controller']['image']
+                wm_img = components['reana-workflow-monitor']['image']
+                mb_img = components['reana-message-broker']['image']
+                wfe_img = components['reana-workflow-engine-yadage']['image']
+
+                # Render the template using given backend config parameters
+                cluster_conf = template.render(backend_conf_parameters,
+                                               SERVER_IMAGE=server_img,
+                                               JOB_CONTROLLER_IMAGE=jc_img,
+                                               WORKFLOW_CONTROLLER_IMAGE=wfc_img,
+                                               WORKFLOW_MONITOR_IMAGE=wm_img,
+                                               MESSAGE_BROKER_IMAGE=mb_img,
+                                               WORKFLOW_ENGINE_IMAGE=wfe_img,
```

Comment:
We can have more than one workflow engine, like `CWL` how are we going to handle it this way?

Reply:
Yes, good and valid question ;)
Let's talk about this offline first and sketch something?